pkg/infra/baseagent.go

```diff
@@ -93,8 +95,15 @@ func (a *Base) setHP(newHP int) {
 	a.hp = newHP
 }
 
+// Modeled as a first order system step answer (see documentation for more information)
 func (a *Base) updateHP(foodTaken float64) {
-	a.hp = int(math.Min(100, float64(a.hp)+foodTaken))
+        hpChange := a.tower.healthInfo.Width*(1-math.Pow(math.E, -foodTaken/a.tower.healthInfo.Tau))
+	if a.hp >= a.tower.healthInfo.WeakLevel {
+		a.hp = a.hp + int(hpChange)
+	} else {
+		a.hp = int(math.Min(float64(a.tower.healthInfo.HPReqCToW), float64(a.hp)+hpChange))
```

Comment:
What is it you want this line to do, as your just setting the hp to HPReqCToW (2) each time as this is always going to be the mimimum value??

Reply:
Do you want something like `a.hp = a.hp + int(math.Min(float64(a.tower.healthInfo.HPReqCToW), hpChange))` instead?

Reply:
This is a mistake, It should be `math.Min(float64(t.healthInfo.HPCritical+a.tower.healthInfo.HPReqCToW), float64(a.hp)+hpChange)`. Let me commit that